File: BarberApp_Nhom6_Be/app/models/shop.py
from __future__ import annotations
from sqlalchemy import Integer, String, Boolean, DateTime, Time, DECIMAL
from sqlalchemy.orm import Mapped, mapped_column, relationship
from datetime import datetime, time
from decimal import Decimal
from app.models.base import Base

class Shop(Base):
    __tablename__ = "shops"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(100))
    address: Mapped[str] = mapped_column(String(255))
    lat: Mapped[Decimal | None] = mapped_column(DECIMAL(10, 6))
    lng: Mapped[Decimal | None] = mapped_column(DECIMAL(10, 6))
    phone: Mapped[str | None] = mapped_column(String(20))
    open_time: Mapped[time | None] = mapped_column(Time)
    close_time: Mapped[time | None] = mapped_column(Time)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[datetime | None] = mapped_column(DateTime, default=datetime.utcnow)

    stylists: Mapped[list["Stylist"]] = relationship(back_populates="shop")
    bookings: Mapped[list["Booking"]] = relationship(back_populates="shop")

File: BarberApp_Nhom6_Fe/barberapp_nhom6_fe/lib/models/shop.dart
class Shop {
  final int id;
  final String name;
  final String address;
  final String? phone;
  final bool isActive;
  final double? lat;
  final double? lng;
  final String? openTime;
  final String? closeTime;

  Shop({
    required this.id,
    required this.name,
    required this.address,
    this.phone,
    required this.isActive,
    this.lat,
    this.lng,
    this.openTime,
    this.closeTime,
  });

  factory Shop.fromJson(Map<String, dynamic> json) {
    return Shop(
      id: json['id'] as int,
      name: json['name'] ?? '',
      address: json['address'] ?? '',
      phone: json['phone'],
      isActive: json['is_active'] ?? true,
      lat: (json['lat'] != null)
          ? double.tryParse(json['lat'].toString())
          : null,
      lng: (json['lng'] != null)
          ? double.tryParse(json['lng'].toString())
          : null,
      openTime: json['open_time'],
      closeTime: json['close_time'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'address': address,
    'phone': phone,
    'is_active': isActive,
    'lat': lat,
    'lng': lng,
    'open_time': openTime,
    'close_time': closeTime,
  };
}
